res/text.f.glsl

```diff
@@ -24,15 +24,21 @@ uniform sampler2D mask;
 
 void main()
 {
-    if (backgroundPass != 0) {
+    if (pass == 0) {
+        // Background.
         if (bg.a == 0.0)
             discard;
 
         alphaMask = vec4(1.0);
         color = vec4(bg.rgb, 1.0);
-    } else {
+    } else if (pass == 1) {
+        // Regular text glyphs.
         vec3 textColor = texture(mask, TexCoords).rgb;
         alphaMask = vec4(textColor, textColor.r);
         color = vec4(fg, 1.0);
+    } else {
+        // Colored glyphs.
+        color = texture(mask, TexCoords);
```

Comment:
You should only use the alpha for `alphaMask`, but in the same time you should return `color = color(color.rgb, 1.0)` to solve mentioned issue. So it should look something like that, I guess.
```glsl
vec4 glyphColor = texture(mask, TexCoords);
alphaMask = vec4(color.a);
color = vec4(glyphColor.rgb, 1.0);
```

Reply:
Wow, good catch. Thanks!